src/layouts/ReviewRequest/components/Comments/CommentsDrawer.tsx

```diff
@@ -0,0 +1,174 @@
+import {
+  Box,
+  Drawer,
+  DrawerBody,
+  DrawerCloseButton,
+  DrawerContent,
+  DrawerHeader,
+  DrawerOverlay,
+  Stack,
+  UseDisclosureReturn,
+  Text,
+  Flex,
+  Divider,
+  HStack,
+  useDisclosure,
+  Skeleton,
+} from "@chakra-ui/react"
+import { IconButton } from "@opengovsg/design-system-react"
+import { PropsWithChildren, useEffect } from "react"
+import { BiCommentDetail } from "react-icons/bi"
+
+import {
+  useGetComments,
+  useUpdateComments,
+  useUpdateReadComments,
+} from "hooks/commentsHooks"
+
+import { getAxiosErrorMessage } from "utils/axios"
+import { getDateTimeFromUnixTime } from "utils/date"
+
+import { EmptyChatImage } from "assets/images/EmptyChatImage"
+import { CommentProps } from "types/comments"
+
+import { CommentFormProps, SendCommentForm } from "./SendCommentForm"
+
+export interface CommentItemProps {
+  commenterName: string
+  commentTime: number
+  isNew: boolean
+}
+
+const CommentItem = ({
+  commenterName,
+  commentTime,
+  isNew,
+  children,
+}: PropsWithChildren<CommentItemProps>): JSX.Element => {
+  const { date, time } = getDateTimeFromUnixTime(commentTime)
+  return (
+    <Box backgroundColor={isNew ? "blue.100" : "blue.50"} p="1rem" mb="1rem">
+      <HStack
+        fontSize="0.625rem"
+        color="text.helper"
+        justifyContent="space-between"
+        pb="0.5rem"
+      >
+        <Text overflow="hidden" textOverflow="ellipsis" whiteSpace="nowrap">
+          {commenterName}
+        </Text>
+        <Text whiteSpace="nowrap">{`${date}, ${time}`}</Text>
+      </HStack>
+      {children}
+    </Box>
+  )
+}
+
+export type CommentsDrawerProps = Pick<
+  UseDisclosureReturn,
+  "onClose" | "isOpen"
+>
+
+export const CommentsDrawer = ({ siteName, requestId }: CommentProps) => {
+  const {
+    isOpen: isCommentsOpen,
+    onOpen: onCommentsOpen,
+    onClose: onCommentsClose,
+  } = useDisclosure()
+
+  const {
+    data: commentsData,
+    isLoading: isCommentsLoading,
+    refetch: refetchRecentNotificationData,
+  } = useGetComments({
+    siteName,
+    requestId,
+  })
+
+  const { mutateAsync: updateReadComments } = useUpdateReadComments()
+  const {
+    mutateAsync: updateNotifications,
+    error: updateNotificationsError,
+  } = useUpdateComments()
+
+  const handleUpdateNotifications = async ({ comment }: CommentFormProps) => {
+    await updateNotifications({ siteName, requestId, message: comment })
+  }
+
+  useEffect(() => {
+    refetchRecentNotificationData()
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Was because of typescript not allowing an empty dependency array for useEffect - but it's been removed in https://github.com/isomerpages/isomercms-frontend/commit/01641ddb3d79e2a9fbe6bb786c1e8f068ce27956